tm2/pkg/bft/wal/wal.go

```diff
@@ -499,7 +499,15 @@ func (enc *WALWriter) Write(v TimedWALMessage) error {
 
 	line64 := base64stdnp.EncodeToString(line)
 	line64 += "\n"
-	_, err := enc.wr.Write([]byte(line64))
+
+	input := []byte(line64)
+
+	n, err := enc.wr.Write(input)
+
+	if len(input) != n {
+		return fmt.Errorf("could not write msg: %+v bytes", input)
+	}
+
```

Comment:
Note that this is a violation of the "contract" of `io.Writer`:

>    Write writes len(p) bytes from p to the underlying data stream. It returns
    the number of bytes written from p (0 <= n <= len(p)) and any error
    encountered that caused the write to stop early. **Write must return a non-nil
    error if it returns n < len(p).** Write must not modify the slice data,
    even temporarily.

If you know of a writer that could return a nil error and not have written the full buffer, it should be fixed.

So I think the check is a bit redundant, and this was better before...

Reply:
done